src/otx/core/data/dataset/detection.py

```diff
@@ -49,8 +49,9 @@ def _get_item_impl(self, index: int) -> DetDataEntity | None:
                 bboxes,
                 format=tv_tensors.BoundingBoxFormat.XYXY,
                 canvas_size=img_shape,
+                dtype=torch.float32,
             ),
-            labels=torch.as_tensor([ann.label for ann in bbox_anns]),
+            labels=torch.as_tensor([ann.label for ann in bbox_anns], dtype=torch.int32),
```

Comment:
Could you align it with iseg to int64 if it works?

Reply:
I think it should work but is it needed to set it to torch.int64? I think there is no case to use more labels than upper limit of int32.

Reply:
It seems that some code in detection head is using `torch.long` for label calculation and need to align it with that. maybe this is why iseg uses torch.int64. Anyway, I updated it to `torch.long`. Thanks.

Reply:
I think so, otx seems to use double and just aligning it :)